### PR TITLE
Improve use of palette macro and use PAL_* constants

### PR DESCRIPTION
**Pull request by Idain**

### CODE REVIEW

File: engine/battle_anims/bg_effects.asm [hunk 1]
```diff
@@ -2567,8 +2567,8 @@ BGEffects_LoadBGPal0_OBPal1:
 	ld b, a
 	ld c, $1
 	call CopyPals
-	ld hl, wOBPals2 palette 1
-	ld de, wOBPals1 palette 1
+	ld hl, wOBPals2 palette PAL_BATTLE_OB_PLAYER
```
Comment (Rangi42) [inline engine/battle_anims/bg_effects.asm:2570]: `ld hl, wBGPals2` and `ld de, wBGPals1` above should have `palette PAL_BATTLE_BG_PLAYER`, and the whole function should be renamed to `BGEffects_LoadPlayerPals`.

File: engine/battle_anims/bg_effects.asm [hunk 2]
```diff
@@ -2589,8 +2589,8 @@ BGEffects_LoadBGPal1_OBPal0:
 	ld a, h
 	push bc
 	push af
-	ld hl, wBGPals2 palette 1
-	ld de, wBGPals1 palette 1
+	ld hl, wBGPals2 palette PAL_BATTLE_BG_ENEMY
```
Comment (Rangi42) [inline engine/battle_anims/bg_effects.asm:2592]: `ld hl, wOBPals2` and `ld de, wOBPals1` above should have `palette PAL_BATTLE_OB_ENEMY`, and the whole function should be renamed to `BGEffects_LoadEnemyPals`.

File: engine/tilesets/timeofday_pals.asm [hunk 3]
```diff
@@ -168,7 +168,7 @@ FillWhiteBGColor:
 	ld e, a
 	ld a, [hli]
 	ld d, a
-	ld hl, wBGPals1 + 1 palettes
+	ld hl, wBGPals1 palette 1
```
[the remark this answers is not in the document]
Comment (Idain) [reply engine/tilesets/timeofday_pals.asm:171]: I feel this should go in its own PR.

**Comment by Idain** [comment]
@vulcandth, do you have another suggestion?

File: engine/tilesets/timeofday_pals.asm (hunk 3 above)
[the remark this answers is not in the document]
Comment (vulcandth) [reply engine/tilesets/timeofday_pals.asm]: Nah we do it now so we don't forget later 😄

**Review by vulcandth — APPROVED** [review]
Looks good; will merge soon when I have got the time to merge to pokegold as well.

**Pull request merged**